app/scripts/routes/FreeTrial.jsx

```diff
@@ -6,14 +6,37 @@ export default class FreeTrial extends React.Component {
     super(props);
     this.state = {
       inputvalue: '',
+      instrument: '',
+      style: '',
+      experience: '',
+      haveInstrument: false,
+      address1: '',
+      address2: '',
+      city: '',
+      state: '',
+      zip: '',
+      days: {
+        monday: false,
+        tuesday: false,
+        wednesday: false,
+        thursday: false,
+        friday: false,
+        saturday: false,
+        sunday: false
+      },
+      allergies: '',
+      specialNeeds: ''
     };
     this.handleChange = this.handleChange.bind(this);
   }
 
   handleChange(event) {
+    console.log("event",event.target);
+    const {id, value}=event.target
     this.setState({
-      inputvalue: event.target.value,
+      [id]: value,
```

Comment:
Does this work? I've never seen it written like this, but also still pretty new to React.

Reply:
This is what Jesus and I came up with at the last meeting, and it seems to work.